pettingzoo/test/api_test.py

```diff
@@ -382,15 +383,18 @@ def play_test(env, observation_0, num_cycles):
         if not env.agents:
             break
 
-        if isinstance(env.observation_space(agent), gymnasium.spaces.Box):
-            assert env.observation_space(agent).dtype == prev_observe.dtype
         assert env.observation_space(agent).contains(
             prev_observe
         ), "Out of bounds observation: " + str(prev_observe)
 
-        assert env.observation_space(agent).contains(
-            prev_observe
-        ), "Agent's observation is outside of it's observation space"
+        if isinstance(env.observation_space(agent), gymnasium.spaces.Box):
+            if isinstance(prev_observe, dict) and "observation" in prev_observe:
+                assert (
+                    env.observation_space(agent).dtype
+                    == prev_observe["observation"].dtype
+                )
+            else:
+                assert env.observation_space(agent).dtype == prev_observe.dtype
```

Comment:
We try to keep the tests as never-nesting as possible, so syntactically this can be fixed.

But that aside, how can `prev_observe` be a dict when the observation space is a Box?

Reply:
I understand that some envs may have a dict space, but why are we checking if prev_obsrevation is a dict (line 391), when we've checked that the observation space is already a Box (line 390)?

Reply:
I'll look at this again in the near future, I still don't understand why we're checking the prev_observation space when we also check the whole observation space in general.